week-4/E-arrays-of-objects/exercise-2.js

```diff
@@ -7,50 +7,73 @@ Each destination has a name, a distance from Glasgow, and a list of transportati
 */
 
-
 var destination1 = {
-    destinationName: "Edinburgh",
-    distanceKms: 80,
-    transportations: ["car", "bus", "train"]
+  destinationName: "Edinburgh",
+  distanceKms: 80,
+  transportations: ["car", "bus", "train"]
 };
 
 var destination2 = {
-    destinationName: "London",
-    distanceKms: 650,
-    transportations: ["car", "bus", "train"]
+  destinationName: "London",
+  distanceKms: 650,
+  transportations: ["car", "bus", "train"]
 };
 
 var destination3 = {
-    destinationName: "Paris",
-    distanceKms: 900,
-    transportations: ["train", "plane"]
+  destinationName: "Paris",
+  distanceKms: 900,
+  transportations: ["train", "plane"]
 };
 
 var destination4 = {
-    destinationName: "Dublin",
-    distanceKms: 350,
-    transportations: ["plane", "ferry"]
+  destinationName: "Dublin",
+  distanceKms: 350,
+  transportations: ["plane", "ferry"]
 };
 
-var travelDestinations = [destination1, destination2, destination3, destination4];
+var travelDestinations = [
+  destination1,
+  destination2,
+  destination3,
+  destination4
+];
 
 /* 
 DO NOT EDIT ANYTHING ABOVE THIS LINE
 WRITE YOUR CODE BELOW
 */
 
+var destinationNamesWithin500Kms = travelDestinations
+  .filter(destination => destination.distanceKms <= 500)
+  .map(destination => destination.destinationName); // Complete here
 
-var destinationNamesWithin500Kms = // Complete here
-
-var destinationNameReachableByFerry = // Complete here
-
-var destinationNamesMoreThan300KmsAwayByTrain = // Complete here (PRINT THE RESULT IN THE CONSOLE USING FOREACH)
-
+console.log(destinationNamesWithin500Kms);
+var destinationNameReachableByFerry = travelDestinations
+  .filter(destination => destination.transportations.includes("ferry"))
+  .map(destination => destination.destinationName); // Complete here
 
+var destinationNamesMoreThan300KmsAwayByTrain = [];
+travelDestinations.forEach(destination => {
+  if (
+    destination.distanceKms > 300 &&
+    destination.transportations.includes("train")
+  ) {
+    destinationNamesMoreThan300KmsAwayByTrain.push(destination.destinationName);
+  }
+}); // Complete here (PRINT THE RESULT IN THE CONSOLE USING FOREACH)
 /*
 DO NOT EDIT ANYTHING BELOW THIS LINE
 */
 
-console.log("Question 1) Expected result: Edinburgh,Dublin, actual result: " + destinationNamesWithin500Kms);
-console.log("Question 2) Expected result: Dublin, actual result: " + destinationNameReachableByFerry);
-console.log("Question 3) Expected result: London,Paris, actual result: " + destinationNamesMoreThan300KmsAwayByTrain);
+console.log(
+  "Question 1) Expected result: Edinburgh,Dublin, actual result: " +
+    destinationNamesWithin500Kms
+);
+console.log(
+  "Question 2) Expected result: Dublin, actual result: " +
+    destinationNameReachableByFerry
+);
+console.log(
+  "Question 3) Expected result: London,Paris, actual result: " +
+    destinationNamesMoreThan300KmsAwayByTrain
+);
```

Comment:
Well done! 3 good answers here - nicely formatted, which makes it easy to read and easy to review! 😂